app/scripts/controllers/app-state-controller.test.ts

```diff
@@ -43,680 +46,648 @@ const extensionMock = {
-        timeout,
-      );
+    it('should update the timeoutMinutes with the autoLockTimeLimit', async () => {
+      await withController(({ controller, controllerMessenger }) => {
+        const timeout = Date.now();
+
+        controllerMessenger.publish(
+          'PreferencesController:stateChange',
+          {
+            preferences: { autoLockTimeLimit: timeout },
+          } as unknown as PreferencesControllerState,
+          [],
+        );
+
+        expect(controller.state.timeoutMinutes).toStrictEqual(timeout);
+      });
     });
   });
 
   describe('isManifestV3', () => {
-    it('creates alarm when isManifestV3 is true', () => {
+    it('creates alarm when isManifestV3 is true', async () => {
       mockIsManifestV3.mockReturnValue(true);
-      ({ appStateController } = createAppStateController());
-
-      const timeout = Date.now();
-      controllerMessenger.publish(
-        'PreferencesController:stateChange',
-        {
-          preferences: { autoLockTimeLimit: timeout },
-        } as unknown as PreferencesControllerState,
-        [],
-      );
-      const spy = jest.spyOn(
-        appStateController as unknown as { _resetTimer: () => void },
-        '_resetTimer',
-      );
-      appStateController.setLastActiveTime();
-
-      expect(spy).toHaveBeenCalled();
-      expect(extensionMock.alarms.clear).toHaveBeenCalled();
-      expect(extensionMock.alarms.onAlarm.addListener).toHaveBeenCalled();
+      await withController(({ controller, controllerMessenger }) => {
+        const timeout = Date.now();
+        controllerMessenger.publish(
+          'PreferencesController:stateChange',
+          {
+            preferences: { autoLockTimeLimit: timeout },
+          } as unknown as PreferencesControllerState,
+          [],
+        );
+        controller.setLastActiveTime();
+
+        expect(extensionMock.alarms.clear).toHaveBeenCalled();
+        expect(extensionMock.alarms.onAlarm.addListener).toHaveBeenCalled();
+      });
     });
   });
 
   describe('AppStateController:getState', () => {
-    it('should return the current state of the property', () => {
-      expect(
-        appStateController.store.getState().recoveryPhraseReminderHasBeenShown,
-      ).toStrictEqual(false);
-      expect(
-        controllerMessenger.call('AppStateController:getState')
-          .recoveryPhraseReminderHasBeenShown,
-      ).toStrictEqual(false);
+    it('should return the current state of the property', async () => {
+      await withController(({ controller, controllerMessenger }) => {
+        expect(
+          controller.state.recoveryPhraseReminderHasBeenShown,
+        ).toStrictEqual(false);
+
+        expect(
+          controllerMessenger.call('AppStateController:getState')
+            .recoveryPhraseReminderHasBeenShown,
+        ).toStrictEqual(false);
+      });
     });
   });
 
   describe('AppStateController:stateChange', () => {
```

Comment:
Similar as above, what are your thoughts on removing the tests for `stateChange` entirely?